routes/runtime/github.rb

```diff
@@ -54,6 +54,37 @@ class CloverRuntime
           end
         }
       end
+
+      # YYYY: Create bucket if not exists
+
+      r.post true do
+        key = r.params["key"]
+        version = r.params["version"]
+        size = Integer(r.params["cacheSize"])
+        fail CloverError.new(400, "InvalidRequest", "Wrong parameters") if key.nil? || version.nil? || size.nil?
+
+        unless (scope = runner.workflow_job&.dig("head_branch"))
+          # YYYY: If the webhook not delivered yet, we can try to get the branch from the API
+          Clog.emit("The runner does not have a workflow job") { {no_workflow_job: {ubid: runner.ubid, repository_ubid: repository.ubid}} }
+          fail CloverError.new(400, "InvalidRequest", "No workflow job data available")
+        end
+
+        entry = GithubCacheEntry.create_with_id(repository_id: runner.repository.id, key: key, version: version, size: size, scope: scope, created_by: runner.id)
+
+        upload_id = repository.blob_storage_client.create_multipart_upload(bucket: repository.bucket_name, key: entry.blob_key).upload_id
+        entry.update(upload_id: upload_id)
+
+        max_chunk_size = 32 * 1024 * 1024 # 32MB
+        presigned_urls = (1..size.fdiv(max_chunk_size).ceil).map do
+          repository.url_presigner.presigned_url(:upload_part, bucket: repository.bucket_name, key: entry.blob_key, upload_id: upload_id, part_number: _1)
+        end
+
+        {
+          cacheId: 1,
```

Comment:
Should this be 1 in all cases? Seems like a weird protocol to return fixed value from endpoint.

Reply:
I've added it, ensuring not to unnecessarily alter the client code. It returns different values from the default GitHub Cache server, typically small integers. However, since we already use `uploadId` for S3 compatibility, we don't utilize `cacheId` at all. I found it weird too, so I decided to bite the bullet and refactor the client code.

> The cache client uses the cacheId in some logics. This is unnecessary for our forked version since we already have uploadId. However, to avoid extensive modifications to the client code, a fixed cacheId is returned for each request.

Reply:
Sounds good, maybe add a comment saying that this value is only there to conform to the protocol and is not being used actually.